docs/kb/semgrep-ci/jenkins-diff-scans.md

```diff
@@ -0,0 +1,185 @@
+        sh '''docker pull semgrep/semgrep && \
+            docker run \
+            -e SEMGREP_APP_TOKEN=$SEMGREP_APP_TOKEN \
+            -e SEMGREP_REPO_NAME=$SEMGREP_REPO_NAME \
+            -v "$(pwd):$(pwd)" --workdir $(pwd) \
+            semgrep/semgrep semgrep ci '''
+      }
+    }
+  }
+}
+```
+
+## Adding diff-aware scans
+
+Semgrep diff-aware scans can be set up in several different ways using Jenkins. This example sets up a Multibranch Pipeline using `when` conditions in the Jenkinsfile, using a full scan configuration based on the example in the preceding section.
+
+The intent of the configuration is to run full scans on the default branch and diff-aware scans on PR branches. Using a Multibranch pipeline provides access to useful variables for the configuration.
+
+### Create the Jenkinsfile
+
+Add the following to a Jenkinsfile in the root of the repository. This code snippet uses Jenkins declarative syntax and runs Semgrep in Docker.
+
+```bash
+pipeline {
+  agent any
+  environment {
+    // Required for a Semgrep Cloud Platform-connected scan:
+    SEMGREP_APP_TOKEN = credentials('SEMGREP_APP_TOKEN')
+    // Set repo name to expected format
+    SEMGREP_REPO_NAME = env.GIT_URL.replaceFirst(/^https:\/\/github.com\/(.*)$/, '$1')
+  }
+  stages {
+    stage('semgrep-diff-scan') {
+      when {
+        branch "PR-*"
+      }
+      steps {
+        sh '''git fetch --no-tags --force --progress -- $GIT_URL +refs/heads/$CHANGE_TARGET:refs/remotes/origin/$CHANGE_TARGET
+              git checkout -b $CHANGE_TARGET origin/$CHANGE_TARGET
+              git checkout $GIT_BRANCH
+           '''
+        sh '''docker pull semgrep/semgrep && \
+            docker run \
+            -e SEMGREP_APP_TOKEN=$SEMGREP_APP_TOKEN \
+            -e SEMGREP_REPO_NAME=$SEMGREP_REPO_NAME \
+            -e SEMGREP_BASELINE_REF=$(git merge-base $GIT_BRANCH $CHANGE_TARGET) \
+            -e SEMGREP_PR_ID = "${env.CHANGE_ID}"
+            -v "$(pwd):$(pwd)" --workdir $(pwd) \
+            semgrep/semgrep semgrep ci '''
+      }
+    }
+    stage('semgrep-scan') {
+      when {
+        branch "main"
+      }
+      steps {
+        sh '''docker pull semgrep/semgrep && \
+            docker run \
+            -e SEMGREP_APP_TOKEN=$SEMGREP_APP_TOKEN \
+            -e SEMGREP_REPO_NAME=$SEMGREP_REPO_NAME \
+            -v "$(pwd):$(pwd)" --workdir $(pwd) \
+            semgrep/semgrep semgrep ci '''
+      }
+    }
+  }
+  post {
+    // Clean after build
+    always {
+      cleanWs()
+    }
+  }
+}
+```
+
+The Jenkinsfile defines two Semgrep stages, each of which is run for certain branches: a diff-aware scan for PR branches, and a full scan for the main branch. This diff-aware scan configuration uses a computed merge base, rather than setting the merge base to the default branch. This is more similar to how Semgrep performs in GitHub actions. Setting the SEMGREP_REPO_NAME and SEMGREP_PR_ID allows Semgrep to identify the connected project and related PR.
```

Comment:
```suggestion
The Jenkinsfile defines two Semgrep stages: one stage runs a diff-aware scan for PR or MR branches, while a second stage runs a full scan for the main branch. The diff-aware scan configuration uses a computed merge base, rather than setting the merge base to the default branch. This is similar to how Semgrep runs in GitHub actions. Setting the SEMGREP_REPO_NAME and SEMGREP_PR_ID allows Semgrep to identify the connected project and related PR.
```

Reply:
This change wouldn't be correct. This guide is specifically for setting up GitHub, which is PR branches. The branches are called PR in Jenkins as well.